diff --git a/wysiwyg.module b/wysiwyg.module
index 22130ea..aba3920 100644
--- a/wysiwyg.module
+++ b/wysiwyg.module
@@ -348,7 +348,7 @@ function wysiwyg_load_editor($profile) {
             drupal_add_js($init, array('type' => 'inline') + $default_library_options);
           }
           else {
-            drupal_add_js(file_create_url($uri), $default_library_options);
+            drupal_add_js($uri, $default_library_options);
           }
         }
       }
